controllers/student.go

```diff
@@ -0,0 +1,89 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"kwoc-backend/middleware"
+	"kwoc-backend/utils"
+	"net/http"
+
+	"kwoc-backend/models"
+
+	"gorm.io/gorm"
+)
+
+type RegisterStudentReqFields struct {
+	Username string `json:"username"`
+	Name     string `json:"name"`
+	Email    string `json:"email"`
+	College  string `json:"college"`
+}
+
+func (dbHandler *DBHandler) RegisterStudent(w http.ResponseWriter, r *http.Request) {
+	db := dbHandler.db
+	var reqFields = RegisterStudentReqFields{}
+
+	err := json.NewDecoder(r.Body).Decode(&reqFields)
+	if err != nil {
+		utils.LogErrAndRespond(r, w, err, "Error decoding JSON body.", http.StatusBadRequest)
+		return
+	}
+
+	// Check if the JWT login username is the same as the student's given username
+	login_username := r.Context().Value(middleware.LoginCtxKey(middleware.LOGIN_CTX_USERNAME_KEY)).(string)
+
+	if reqFields.Username != login_username {
+		utils.LogWarn(
+			r,
+			fmt.Sprintf(
+				"POSSIBLE SESSION HIJACKING\nJWT Username: %s, Given Username: %s",
+				login_username,
+				reqFields.Username,
+			),
+		)
+
+		w.WriteHeader(http.StatusUnauthorized)
+		fmt.Fprint(w, "Login username and given username do not match.")
+		return
+	}
+
+	// Check if the student already exists in the db
+	student := models.Student{}
+	tx := db.
+		Table("students").
+		Where("username = ?", reqFields.Username).
```

Comment:
would checking email instead of username help in solving the `changed username due to falana dimka` issue?

Reply:
They can change email too /shrug. Also the email is given manually and can be wrong whereas the username is obtained from the Github api itself.